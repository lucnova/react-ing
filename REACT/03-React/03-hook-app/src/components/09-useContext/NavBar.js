import React from 'react';
import { Link, NavLink } from 'react-router-dom';

export const NavBar = () => {
	return (
		<nav className="navbar navbar-expand-lg navbar-dark bg-dark px-3">
			<Link to="/" className="navbar-brand">
				UseContext
			</Link>

			<div className="collapse navbar-collapse">
				<ul className="navbar-nav mr-auto">
					<NavLink exact to="/" className="nav-item nav-link" activeClassName="active">
						Home
					</NavLink>
					<NavLink exact to="/about" className="nav-item nav-link" activeClassName="active">
						About
					</NavLink>
					<NavLink exact to="/login" className="nav-item nav-link" activeClassName="active">
						Login
					</NavLink>
				</ul>
			</div>
		</nav>
	);
};
