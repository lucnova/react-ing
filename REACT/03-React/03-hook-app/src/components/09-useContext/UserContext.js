// * Higher Order Component
const { createContext } = require("react");

// * Exportarlo
export const UserContext = createContext(null);