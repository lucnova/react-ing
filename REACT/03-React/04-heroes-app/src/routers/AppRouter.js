import React from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { DCPage } from '../components/DC/DCPage';
import { MarvelPage } from '../components/MARVEL/MarvelPage';
import { NavBar } from '../components/UI/NavBar';

export const AppRouter = () => {
	return (
		<Router>
			<div>
				<NavBar />

				<Switch>
					<Route exact path="/dc" component={DCPage} />
					<Route exact path="/marvel" component={MarvelPage} />
				</Switch>
			</div>
		</Router>
	);
};
