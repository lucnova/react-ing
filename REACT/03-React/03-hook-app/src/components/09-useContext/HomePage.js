import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const HomePage = () => {
	const userContext = useContext(UserContext);
	const { user } = userContext;

	return (
		<>
			<h1>Home Page</h1>
			<hr />

			<pre>{JSON.stringify(user, null, 4)}</pre>
		</>
	);
};
