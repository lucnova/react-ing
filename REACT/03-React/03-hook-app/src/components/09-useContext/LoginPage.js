import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginPage = () => {
	const { setUser } = useContext(UserContext);

	return (
		<>
			<h1>Login Page</h1>
			<hr />

			<button
				className="btn btn-primary"
				onClick={() =>
					setUser({
						id: '123ABC',
						username: 'lucnova',
						email: 'alanugb@gmail.com',
					})
				}
			>
				Login
			</button>

			<button className="btn btn-warning" onClick={() => setUser({})}>
				Logout
			</button>
		</>
	);
};
