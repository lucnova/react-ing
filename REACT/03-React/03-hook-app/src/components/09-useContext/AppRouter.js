import React from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { AboutPage } from './AboutPage';
import { HomePage } from './HomePage';
import { LoginPage } from './LoginPage';
import { NavBar } from './NavBar';

export const AppRouter = () => {
	// * El Switch devuelve lo primero que se indique; si pones Route path='/', no importa si te dirijes a /about;
	//       coinciden el / y es suficiente para el Switch,
	//      Para hacerlo exacto, colocar 'exact' como propiedad.
	return (
		<>
			<Router>
				<div>
					<NavBar />
					<Switch>
						<Route exact path="/home" component={HomePage} />
						<Route exact path="/login" component={LoginPage} />
						<Route exact path="/about" component={AboutPage} />
						<Route exact path="/" component={HomePage} />
					</Switch>
				</div>
			</Router>
		</>
	);
};
