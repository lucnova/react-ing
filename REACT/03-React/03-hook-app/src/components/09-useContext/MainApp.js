import React from 'react';
import { AppRouter } from './AppRouter';
import { UserContext } from './UserContext';

// * Establecer el contexto a todo el componente principal.

export const MainApp = () => {
	const user = {
		id: '12ABC',
		username: 'lucnova',
		email: 'nlnovae@gmail.com',
	};

	return (
		<UserContext.Provider value={user}>
			<AppRouter />
		</UserContext.Provider>
	);
};
