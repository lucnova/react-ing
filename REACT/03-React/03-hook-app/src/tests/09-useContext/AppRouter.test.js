import { mount } from 'enzyme';
import { AppRouter } from '../../components/09-useContext/AppRouter';
import { UserContext } from '../../components/09-useContext/UserContext';

// * Este componente es probado al momento solo con snapshot (Aun no esta optmizado.)
//      Ocuparemos probar los argumentos enviados en el router luego.

const user = {
	id: 'TEST',
	username: 'lucnova',
	email: 'alanugb@gmail.com',
};

describe('Testing on AppRouter', () => {
	const wrapper = mount(
		<UserContext.Provider
			value={{
				user,
			}}
		>
			<AppRouter />
		</UserContext.Provider>
	);

	test('should display correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
