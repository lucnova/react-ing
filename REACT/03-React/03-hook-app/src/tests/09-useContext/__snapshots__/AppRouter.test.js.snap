// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on AppRouter should display correctly 1`] = `
<div>
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark px-3"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      UseContext
    </a>
    <div
      className="collapse navbar-collapse"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </ul>
    </div>
  </nav>
  <div
    className="container"
  >
    Array [
      <h1>
        Home Page
      </h1>,
      <hr />,
      <pre>
        {
    "id": "TEST",
    "username": "lucnova",
    "email": "alanugb@gmail.com"
}
      </pre>,
    ]
  </div>
</div>
`;
